pkg/deploy/util/util.go

```diff
@@ -274,6 +285,60 @@ func CopyApiEnvVarToV1EnvVar(in []api.EnvVar) []v1.EnvVar {
 	return out
 }
 
+func CopyPodTemplateSpecToV1PodTemplateSpec(spec *api.PodTemplateSpec) *v1.PodTemplateSpec {
+	copied, err := api.Scheme.DeepCopy(spec)
+	if err != nil {
+		panic(err)
+	}
+	in := copied.(*api.PodTemplateSpec)
+	out := &v1.PodTemplateSpec{}
+	if err := v1.Convert_api_PodTemplateSpec_To_v1_PodTemplateSpec(in, out, nil); err != nil {
+		panic(err)
+	}
+	return out
+}
+
+// HasLatestPodTemplate checks if the provided replication controller has the latest
+// deployment config pod template. If not it will return false and the string diff.
+func HasLatestPodTemplate(dc *deployapi.DeploymentConfig, rc *v1.ReplicationController) (bool, string, error) {
+	configTemplate := CopyPodTemplateSpecToV1PodTemplateSpec(dc.Spec.Template)
+	strippedRc, err := StripOriginLabelsAndAnnotations(rc)
+	if err != nil {
+		return true, "", err
+	}
+	rcTemplate := strippedRc.Spec.Template
+	// there is difference between map[string]string(nil) vs. map[string]string{}
+	if configTemplate.Annotations == nil {
+		configTemplate.Annotations = map[string]string{}
+	}
+	if configTemplate.Labels == nil {
+		configTemplate.Labels = map[string]string{}
+	}
+	if reflect.DeepEqual(configTemplate, rcTemplate) {
+		return true, "", nil
+	}
+	return false, diff.ObjectReflectDiff(configTemplate, rcTemplate), nil
+}
+
+// StripOriginLabelsAndAnnotations removes all origin specific labels from the
+// replication controller pod template.
+// NOTE: Every new label or annotation we add to replication controller template we create
+// has to be stripped down here otherwise the config change trigger will keep
+// triggering.
+func StripOriginLabelsAndAnnotations(in *v1.ReplicationController) (*v1.ReplicationController, error) {
+	copy, err := DeploymentDeepCopyV1(in)
+	if err != nil {
+		return nil, err
+	}
+	delete(copy.Spec.Template.Annotations, deployapi.DeploymentVersionAnnotation)
```

Comment:
Follow the upstream pattern: https://github.com/kubernetes/kubernetes/blob/ede78d9ee7706b5d5dab943e2b56f373308f578e/pkg/controller/deployment/util/deployment_util.go#L281